modal/cls.py

```diff
@@ -460,3 +528,52 @@ def __getattr__(self, k):
 
 
 Cls = synchronize_api(_Cls)
+
+
+class _NO_DEFAULT:
+    def __repr__(self):
+        return "modal.cls._NO_DEFAULT()"
+
+
+class _Parameter:
+    default: Any
+
+    def __init__(self, default: Any):
+        self.default = default
+
+
+def parameter(*, default: Any = _NO_DEFAULT()) -> Any:
+    """Used to specify options for modal.cls parameters, similar to dataclass.field for dataclasses
+    ```
+    class A:
+        a: str = modal.parameter()
+
+    ```
+    """
+    # has to return Any to be assignable to any annotation (https://github.com/microsoft/pyright/issues/5102)
+    return _Parameter(default)
+
+
+def field(*, default: Any = None, init: bool = False) -> Any:
+    """Optional field descriptor for non-parameter fields on a Modal class
+
+    Using this on an annotated field allows *type checkers* and editors to omit the field in
+    the implicit constructor of the class. It has no effect on the actual runtime constructor
```

Comment:
Yeah it's basically only there since type checkers would complain otherwise that the argument is missing in the constructor. Another option that would be closely related to the built-in dataclass syntax is to have an `init: bool=True` argument to `modal.parameter()` that could be to omit it from the constructor. But that might be a little confusing as the parameter would then not be a parameter in the modal sense anymore... Perhaps good enough for people who want to get around type checking issues though...

Reply:
I still think that borrowing the name from data class tools but behaving differently is pretty confusing … but maybe I’m missing something